assets/apps/components/src/Common/CustomPalette.js

```diff
@@ -2,12 +2,25 @@ import { __ } from '@wordpress/i18n';
 import { Button, ExternalLink } from '@wordpress/components';
 import classnames from 'classnames';
 import { globalPaletteColors } from './common';
+import { useState } from '@wordpress/element';
 
 const CustomPalette = ({ title, onChange, activeColor }) => {
 	const focusGlobalColors = () => {
 		wp.customize.control('neve_global_colors').focus();
 	};
 
+	const globalCustomColorsControl = wp.customize.control(
+		'neve_global_custom_colors'
+	);
+	const customColorsState = useState({
+		...globalCustomColorsControl.setting.get(),
+	})[0];
+
+	const customColors = Object.entries(customColorsState)
+		.filter(([key]) => key !== 'flag')
+		.map(([key, value]) => ({ [key]: value.label }))
+		.reduce((acc, cur) => Object.assign(acc, cur), {});
+
```

Comment:
I have suggestions regarding here;
* I wonder, do we need to use a `state`? in L15 . If not, we can remove the `useState` call
* In L19-22, we can replace the separate JS `.filter` `.map` and `.reduce` usages with a single JS loop. That may improve the performance positively.

Apart from that, we can these which get to custom colors to another function if it makes sense to you.